awx/main/credential_plugins/dsv.py

```diff
@@ -0,0 +1,71 @@
+from .plugin import CredentialPlugin
+
+from django.conf import settings
+from django.utils.translation import get_language, to_locale, ugettext_lazy as _
+from thycotic.secrets.vault import SecretsVault
+
+
+dsv_inputs = {
+    'fields': [
+        {
+            'id': 'tenant',
+            'label': _('Tenant'),
+            'help_text': _('The tenant e.g. "ex" when the URL is https://ex.secretservercloud.com'),
+            'type': 'string',
+        },
+        {'id': 'client_id', 'label': _('Client ID'), 'type': 'string'},
+        {
+            'id': 'client_secret',
+            'label': _('Client Secret'),
+            'type': 'string',
+            'secret': True,
+        },
+    ],
+    'metadata': [
+        {
+            'id': 'path',
+            'label': _('Secret Path'),
+            'type': 'string',
+            'help_text': _('The secret path e.g. /test/secret1'),
+        },
+        {
+            'id': 'expr',
+            'label': _('Attribute Expression'),
+            'type': 'string',
+            'help_text': _('A Python expression on the dict representation of the secret e.g. "data.username"'),
+        },
+    ],
+    'required': ['tenant', 'client_id', 'client_secret', 'path'],
+}
+
+if settings.DEBUG:
+    dsv_inputs['fields'].append(
+        {
+            'id': 'url_template',
+            'label': _('URL template'),
+            'type': 'string',
+            'default': 'https://{}.secretsvaultcloud.{}/v1',
+        }
+    )
+
+if to_locale(get_language()) != 'en_US':
```

Comment:
Can you explain what's going on here?

Reply:
Why, a dubious assumption, of course! 😁

The TLD reflects where the cloud instance lives and it defaults to '.com' which is a US deployment.

The assumption is that the user will need to use a non-default TLD for DSV only when AWX is running in a non-US region.

I originally had an override in settings.py but I removed it.

Reply:
It just seems strange, does it not make more sense to just have them enter the base url on its own? If the tld or, god forbid, the url structure needs to change then it would require a code change that wouldn't be picked up until the next Tower release. Seems like something that could get broken pretty easily?